### PR TITLE
Fixed Custom Bastion role unable to access EKS cluster

### PR DESCRIPTION
**Pull request by nfairoza**
1) exposed the controlplane provision role arn in EKSSTACK 2) added policy based on custombastionrole condition 3) exposed the nodegroup ASG name

*Issue #, if available:*  EKS is created ControlPlaneProvisionRole. default bastion role has authorization to assume ControlPlaneProvisionRole to communicate with EKS cluster.
I used custom bastion role, This new custom role is not given sts Assume role access to the ControlPlaneProvisionRole automatically. Thus my bastion host is unable to communicate with EKS cluster.

*Description of changes:*
I added CustomBastionRole condition to iam stack, if it is true then it will add a policy to ControlPlaneProvisionRole which will allow my custom role to be able to assume ControlPlaneProvisionRole .
exposed ControlPlaneProvisionRole and nodegroupASG name, I need them for my quickstart cloudformation template  deveopment which is using EKS QS as submodule.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by nfairoza** [comment]
@vsnyc

File: templates/amazon-eks.template.yaml [hunk 1]
```diff
@@ -279,6 +279,7 @@ Resources:
         BootstrapArguments: !Ref BootstrapArguments
         QSS3BucketName: !Ref QSS3BucketName
         QSS3KeyPrefix: !Ref QSS3KeyPrefix
+        CleanupSecurityGroupDependenciesLambdaArn: !GetAtt FunctionStack.Outputs.CleanupSecurityGroupDependenciesLambdaArn
```
Comment (vsnyc) [inline templates/amazon-eks.template.yaml]: @nfairoza Why is this line necessary? Do you also need to update the `amazon-eks-nodegroup.template.yaml` template?

**Comment by vsnyc** [comment]
@jaymccon Can you look at this PR? Looks like currently, there isn't a way to pass a custom bastion role to the bastion host?
